### PR TITLE
KAFKA-17956: Remove Admin.listShareGroups

### PR DESCRIPTION
**Pull request by AndrewJSchofield**
KIP-1043 introduced `Admin.listGroups` as the way to list all types of groups. As a result, `Admin.listShareGroups` has been removed. This PR is the final step of the removal.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java [hunk 1]
```diff
@@ -3855,48 +3855,6 @@ public DeleteConsumerGroupOffsetsResult deleteConsumerGroupOffsets(
         return new DeleteConsumerGroupOffsetsResult(future.get(CoordinatorKey.byGroupId(groupId)), partitions);
     }
 
-    private static final class ListShareGroupsResults {
-        private final List<Throwable> errors;
-        private final HashMap<String, ShareGroupListing> listings;
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java:3860]: Could you please remove `ShareGroupListing` as well?

File: clients/src/main/java/org/apache/kafka/clients/admin/ListShareGroupsResult.java [hunk 2]
```diff
@@ -1,84 +0,0 @@
...
-import java.util.ArrayList;
-import java.util.Collection;
-
-/**
- * The result of the {@link Admin#listShareGroups(ListShareGroupsOptions)} call.
- * <p>
- * The API of this class is evolving, see {@link Admin} for details.
- */
-@InterfaceStability.Evolving
-public class ListShareGroupsResult {
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/clients/admin/ListShareGroupsResult.java:33]: Could you please remove those classes from [KIP-932](https://cwiki.apache.org/confluence/display/KAFKA/KIP-932%3A+Queues+for+Kafka#KIP932:QueuesforKafka-ListShareGroupsResult)

**Review by chia7712 — COMMENTED** [review]
@AndrewJSchofield thanks for this cleanup!

File: clients/src/main/java/org/apache/kafka/clients/admin/ListShareGroupsResult.java (hunk 2 above)
Comment (AndrewJSchofield) [reply clients/src/main/java/org/apache/kafka/clients/admin/ListShareGroupsResult.java:33]: Done. I have aligned KIP-932 with KIP-1043. This means:
* Removal of `Admin.listShareGroups` and the associated classes.
* Replacement of `ShareGroupState` with the generic `GroupState`.

File: clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java (hunk 1 above)
Comment (AndrewJSchofield) [reply clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java:3860]: Done.

**Comment by AndrewJSchofield** [comment]
Test failure known and fixed by https://issues.apache.org/jira/browse/KAFKA-18078.

**Review by chia7712 — APPROVED** [review]

**Pull request merged**